appengine-java11/guestbook-cloud-firestore/src/main/java/com/example/guestbook/Persistence.java

```diff
@@ -16,27 +16,32 @@
 
 package com.example.guestbook;
 
+// [START gae_java11_firestore_dependencies]
 import com.google.cloud.firestore.Firestore;
 import com.google.cloud.firestore.FirestoreOptions;
+// [END gae_java11_firestore_dependencies]
 import java.util.concurrent.atomic.AtomicReference;
 
 /** Create a persistence connection to your Firestore instance. */
 public class Persistence {
 
-  private static AtomicReference<Firestore> firestore = new AtomicReference<>();
+  private static AtomicReference<Firestore> firestoreRef = new AtomicReference<>();
```

Comment:
You could move this inside the getFirestore method.

Reply:
Do I need the AtomicReference or can I just rebuild the database object? I'm not sure if this needs to be thread-safe.

Reply:
Ask the client library folks.